src/frontend/src/common/queries/auth.ts

```diff
@@ -29,32 +24,28 @@ export const USE_USERDATA = {
   id: "userData",
 };
 
-export interface UserResponse extends APIResponse {
-  group: Group;
-  user: User;
-}
+const mapUserData = (obj: any): User => {
+  return <User>{
```

Comment:
Okay, I spent some time looking into this a bit more, and I don't think this is doing what's intended. Messing around with it locally, I'm seeing some weird stuff.

Here's what I have in my IDE
```
interface MyInterface {
  x: string;
  y: string;
}

const a = <MyInterface>{ // No error, TS is happy as a clam, even though it fails to satisfy MyInterface
  x: "blah"
};
const b: MyInterface = { // Error as expected ("Property 'y' is missing...", what you'd expect)
  x: "blah",
};
const c = <MyInterface>{ // Error, but uh, what? (I'll paste the error below, it's long)
  x: "blah",
  z: "bleh",
};
```

The error for `c` above:
```
Conversion of type '{ x: string; z: string; }' to type 'MyInterface' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'y' is missing in type '{ x: string; z: string; }' but required in type 'MyInterface'.
```

So I don't think the `<Blah>{object}` syntax is a type assertion, but `a` should fail and it didn't, so something weird is definitely going on?

Also, this syntax will blow up if you're writing `.tsx` files because it will try to interpret it as JSX and the `<Blah>` isn't actually a React component tag (or at least it blew up on me at first until I moved my back of the envelope check to a `.ts` file instead of the`.tsx` I had open in my editor when I started messing around)

Reply:
Welp, running the linter/formatter on this code refactored it one way or another 😢